### PR TITLE
NUC-1446: use ZCOUNT to assess ready count

### PR DESCRIPTION
**Pull request by gjesse**

### CODE REVIEW

File: rdbi-recipes/src/main/java/com/lithium/dbi/rdbi/recipes/scheduler/StateDedupedJobSchedulerDAO.java [hunk 1]
```diff
@@ -196,11 +196,11 @@ int deleteJobFromReady(
             @BindArg("job") String job);
 
     @Query(
-            "local readyCount = redis.call('ZCARD', $readyQueue$)\n"+
-            "return readyCount + redis.call('ZCARD', $readyAndRunningQueue$)"
+            "local readyCount = redis.call('ZCOUNT', $readyQueue$, 0, $currentTimeMillis$)\n"+
```
Comment (MattRupeLi) [inline rdbi-recipes/src/main/java/com/lithium/dbi/rdbi/recipes/scheduler/StateDedupedJobSchedulerDAO.java]: in this context, is 0 = epoch?

Comment (gjesse) [reply rdbi-recipes/src/main/java/com/lithium/dbi/rdbi/recipes/scheduler/StateDedupedJobSchedulerDAO.java]: yes

File: rdbi-recipes/src/main/java/com/lithium/dbi/rdbi/recipes/scheduler/MultiChannelSchedulerDAO.java [hunk 2]
```diff
@@ -161,12 +161,13 @@ int inQueue(
             "local count = 0\n" +
             "for i, channel in ipairs(channels) do \n" +
             "  local readyQueue = channel .. \":ready_queue\"\n" +
-            "  count = count + redis.call('ZCARD', readyQueue)\n" +
+            "  count = count + redis.call('ZCOUNT', readyQueue, 0, $currentTimeMillis$)\n" +
```
Comment (2bitoperations) [inline rdbi-recipes/src/main/java/com/lithium/dbi/rdbi/recipes/scheduler/MultiChannelSchedulerDAO.java]: watch out with this - zcount isn't constant time. things could get dicey if you have a whole whole bunch of jobs in ready.

Comment (gjesse) [reply rdbi-recipes/src/main/java/com/lithium/dbi/rdbi/recipes/scheduler/MultiChannelSchedulerDAO.java]: true, but there's no other way to get a count. in omega we cache the result of this for a short time as well.

**Pull request merged**